DietPi-Software | Aria2: Config tweak/fix

msongz opened the pull request:
to fix using 3rd party plugin;
and comment unknown option;


unknown option (`systemctl status aria2.service`):
```
   Loaded: loaded (/etc/systemd/system/aria2.service; disabled; vendor preset: enabled)
   Active: active (running) since Tue 2019-02-12 02:12:35 HKT; 5s ago
 Main PID: 3572 (aria2c)
   CGroup: /system.slice/aria2.service
           └─3572 /usr/bin/aria2c --conf-path=/var/lib/dietpi/dietpi-software/installed/aria2.conf

Feb 12 02:12:35 zerow systemd[1]: Started Aria2 (DietPi).
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: host=localhost
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: cleanup-policy="clean_got"
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: cleanup-percent=90%
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: metalink-servers=7
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: [limit-options]
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [WARN] Unknown option: [unlimit-options]
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [NOTICE] IPv4 RPC: listening on TCP port 6800
Feb 12 02:12:35 zerow aria2c[3572]: 02/12 02:12:35 [NOTICE] IPv6 RPC: listening on TCP port 6800
```

offical ref:
https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-rpc-allow-origin-all

3rd plugin:
https://github.com/acgotaku/BaiduExporter

msongz committed:
add aria2 config option: rpc allow origin all
to fix using 3rd party plugin;
and comment unknown option;

MichaIng commented:
@msongz 
Many thanks for your PR.

<s>On which device/distro version did you test?</s>

On Stretch (and above) we install from APT repo, on Jessie we use an own compiled binary. Perhaps the unknown options (and sectioning in config file) does only work on one of them. Also perhaps the settings names have changed, so instead of removing we could rename them, in case dependant on Jessie vs Stretch.

- [x] But yeah easy to test. Will check on VMs.
_______________
Verified deprecated settings on Jessie, Stretch and Buster, all the same. Those settings seem to be relics from old times 😉:
```
root@VM-Jessie:~# systemctl status aria2
● aria2.service - Aria2 (DietPi)
   Loaded: loaded (/etc/systemd/system/aria2.service; disabled)
   Active: active (running) since Mon 2019-02-11 21:03:42 CET; 26s ago
 Main PID: 6805 (aria2c)
   CGroup: /system.slice/aria2.service
           └─6805 /usr/local/bin/aria2c --conf-path=/var/lib/dietpi/dietpi-software/installed/aria2.conf

Feb 11 21:03:42 VM-Jessie systemd[1]: Started Aria2 (DietPi).
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: host=localhost
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: cleanup-policy=clean_got
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: cleanup-percent=90%
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: metalink-servers=40
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: [limit-options]
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [WARN] Unknown option: [unlimit-options]
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [NOTICE] IPv4 RPC: listening on TCP port 6800
Feb 11 21:03:42 VM-Jessie aria2c[6805]: 02/11 21:03:42 [ERROR] IPv6 RPC: failed to bind TCP port 6800
Feb 11 21:03:42 VM-Jessie aria2c[6805]: Exception: [SocketCore.cc:312] errorCode=1 Failed to bind a socket, cause: Name or service not known
```
- We could even avoid binding to IPv6, if this is disabled. On the other hand doesn't hurt and as fast as IPv6 is enabled, Aria2 will bind to it as well.
- Hmm `host` is a required option for the server performance profile: https://aria2.github.io/manual/en/html/aria2c.html#server-performance-profile
But where does this need to be added 🤔?
**EDIT: https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-server-stat-of**
  - So doesn't seem to be required or default works well.

MichaIng commented:
<s>Ahhh, found where we have those settings from: https://github.com/hzamani/aria2web/blob/master/config/aria2.conf

Is it possible that the Aria2 web UI uses these settings, while the Aria2 binary doesn't know them? Actually no clean solution, instead the web UI should use an own config file... However if those do not hurt, but show those warnings, it is acceptable?</s>
- Ah sorry wrong web UI. We use: https://github.com/ziahamza/webui-aria2
- The above is 7 years old. Perhaps we switched the web UI once but left the old settings in place.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11148,18 +11149,18 @@ max-download-limit=0
 always-resume=true
 auto-file-renaming=false
 file-allocation=none
 
```
The settings below were (from the old web UI) were actually meant to be used either|or. So either you limit download/upload speed or you don't. Like this, I guess the `0` will override the limit above.

`0` is default, leave choice to user to enable limits: https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-max-overall-download-limit

MichaIng committed:
v6.22

+ DietPi-Software | Aria2: Remove deprecated settings (instead of comment only); Remove doubled settings; Add bandwidth limit info

MichaIng commented:
Passed local testing, will merge.
Changelog: https://github.com/Fourdee/DietPi/commit/03394b9338e438aeea3263a1efb9e71e588d14b7

msongz commented:
wow! very detail development! thanks!
